datalad/distribution/subdatasets.py

```diff
@@ -166,17 +173,55 @@ def __call__(
             dataset, check_installed=False, purpose='subdataset reporting')
         refds_path = dataset.path
 
+        # XXX this seems strange, but is tested to be the case -- I'd rather set
+        # `check_installed` to true above and fail
```

Comment:
Well, a `Dataset` instance can exist independently on a physical repository. So `None` as the return value of `subdatasets` makes some sense. On the other hand this most likely is a remainder from a time when there was no `require_dataset(check_installed=...)`. Now it might be more consistent to fail. I have no strong opinion on this one. Feel free to change it.